x-pack/test/security_solution_endpoint_api_int/services/roles_users.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Role } from '@kbn/security-plugin/common';
+
+import { t1Analyst } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/t1_analyst';
+import { t2Analyst } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/t2_analyst';
+import { hunter } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/hunter';
+import { threadIntelligenceAnalyst } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/thread_intelligence_analyst';
+import { socManager } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/soc_manager';
+import { platformEngineer } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/platform_engineer';
+import { endpointOperationsAnalyst } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/endpoint_operations_analyst';
+import { endpointSecurityPolicyManager } from '@kbn/security-solution-plugin/scripts/endpoint/common/roles_users/endpoint_security_policy_manager';
+
+import { FtrProviderContext } from '../ftr_provider_context';
+
+export enum ROLE {
+  t1_analyst = 't1Analyst',
+  t2_analyst = 't2Analyst',
+  analyst_hunter = 'hunter',
+  thread_intelligence_analyst = 'threadIntelligenceAnalyst',
+  detections_engineer = 'detectionsEngineer',
+  soc_manager = 'socManager',
+  platform_engineer = 'platformEngineer',
+  endpoint_operations_analyst = 'endpointOperationsAnalyst',
+  endpoint_security_policy_manager = 'endpointSecurityPolicyManager',
+}
+
+const rolesMapping: { [id: string]: Omit<Role, 'name'> } = {
+  t1Analyst,
+  t2Analyst,
+  hunter,
+  threadIntelligenceAnalyst,
+  socManager,
+  platformEngineer,
+  endpointOperationsAnalyst,
+  endpointSecurityPolicyManager,
+};
+
+export function RolesUsersProvider({ getService }: FtrProviderContext) {
+  const security = getService('security');
+  return {
+    /**
+     * Creates an user with specific values
+     * @param user
+     */
+    async createUser(user: { name: string; roles: string[]; password?: string }) {
```

Comment:
Can you add Return types to the methods here and below

Reply:
https://github.com/elastic/kibana/pull/144710/commits/ec504412153be17a77c025b21ea6f02465fd4f1a